lib/Constants.sol

```diff
@@ -37,26 +37,30 @@ int constant NUM_BINS = 2097152;
 
 uint constant OFFER_MASK = 4294967295;
 
+// The tick range is the largest such that 1.0001^MAX_TICK fits on 128 bits (and thus can be multiplied by volumes)
+int constant MIN_TICK = -887272;
+int constant MAX_TICK = 887272;
+uint constant MIN_RATIO_MANTISSA = 170153974464283981435225617938057077692;
+int constant MIN_RATIO_EXP = 255;
+uint constant MAX_RATIO_MANTISSA = 340256786836388094050805785052946541084;
+int constant MAX_RATIO_EXP = 0;
+uint constant MANTISSA_BITS = 128;
+uint constant MANTISSA_BITS_MINUS_ONE = 127;
+/* 
+With |tick|<=887272 and normalized mantissas on 128 bits, the maximum possible mantissa is 340282295208261841796968287475569060645, so the maximum safe volume before overflow is actually 340282438633630198193436196978374475856. 
 
+The immediate idea is to set MAX_SAFE_VOLUME to `(1<<max_safe_volume_bits)-1`, where `max_safe_volume_bits = 256-MANTISSA_BITS`, for simplicity. But we also have a `ByVolume` API, so we also need the ratios `inbound_amount/outbound_amount` to not correspond to a tick outside of the tick range. 
 
-// +/- 2**20-1 because only 20 bits are examined by the tick->ratio function
-int constant MIN_TICK = -1048575;
-int constant MAX_TICK = 1048575;
-uint constant MIN_RATIO_MANTISSA = 4735129379934731672174804159539094721182826496;
-int constant MIN_RATIO_EXP = 303;
-uint constant MAX_RATIO_MANTISSA = 3441571814221581909035848501253497354125574144;
-int constant MAX_RATIO_EXP = 0;
-uint constant MANTISSA_BITS = 152;
-uint constant MANTISSA_BITS_MINUS_ONE = 151;
-// Maximum volume that can be multiplied by a ratio mantissa
-uint constant MAX_SAFE_VOLUME = 20282409603651670423947251286015;
+Instead of fine-tuning MAX_SAFE_VOLUME to be exactly as big as it can be, we just set `max_safe_volume_bits = 256 - MANTISSA_BITS - 1`.
```

Comment:
Do we gain anything by making `MAX_SAFE_VOLUME` fit within 127 bits?

Reply:
Do you mean as opposed to 128 bits ? We gain the fact that the bounds on offer volume, fillVolume, and takerWants/takerGives are all the same. If it was on 128 bits, we'd need to document the fact that the `ByVolume` functions can sometimes revert.

Reply:
The comment makes is sound like we're lazy and can't be bothered to fine-tune ;-)
But if I understand correctly, that's not the case. Instead, I think you're saying "the actual max is somewhere between 2^127 and 2^128 - 1, but choosing 2^127 - 1 means we can simply restrict volumes to 127 bits and be done with it".

Reply:
yes that's exactly it